code/parachain/frame/composable-maths/src/dex/constant_product.rs

```diff
@@ -318,7 +318,9 @@ pub fn compute_in_given_out_new<T: PerThing>(
 	let power = base.checked_powd(weight_ratio).ok_or(ArithmeticError::Overflow)?;
 	let ratio = power.safe_sub(&Decimal::ONE)?;
 
-	let a_sent = b_i_over_fee.safe_mul(&ratio)?.round_up();
+	// NOTE(connor): for in_given_out to produce compatible results for out_given_in, both must
```

Comment:
We don't really need to have compatibility in results for those , they are separate functionalities. We can keep the greedy rounding behavior